src/main/java/org/gridsuite/directory/server/repository/DirectoryElementRepository.java

```diff
@@ -81,7 +81,8 @@ List<DirectoryElementEntity> findAllStashedElements(@Param("uuids") List<UUID> u
     // We select all stashed elements that do not have a parent, or have a parent that is not deleted, or a parent that is deleted in different operation
     @Query("SELECT e FROM DirectoryElementEntity e " +
             "WHERE e.stashed = true AND " +
-            "(e.isPrivate = false or e.owner = :userId or (e.isPrivate IS NULL AND NOT EXISTS (SELECT 1 FROM DirectoryElementEntity parent WHERE parent.id = e.parentId AND parent.isPrivate = true))) " +
+            "e.owner = :userId AND " +
+            "(e.isPrivate = false or (e.isPrivate IS NULL AND NOT EXISTS (SELECT 1 FROM DirectoryElementEntity parent WHERE parent.id = e.parentId AND parent.isPrivate = true)))" +
```

Comment:
It seems to me that this second line requires the object to be public (private are all excluded). It might explain the problem I encountered.